va/va.c

```diff
@@ -434,149 +596,18 @@ static VAStatus va_openDriver(VADisplay dpy, char *driver_name)
     }
     driver_dir = strtok_r(search_path, ENV_VAR_SEPARATOR, &saveptr);
     while (driver_dir) {
-        void *handle = NULL;
-        char *driver_path = va_getDriverPath(driver_dir, driver_name);
+        char *driver_path = va_getDriverPath(driver_dir, driver_name, add_name_suffix);
         if (!driver_path) {
             va_errorMessage(dpy, "%s L%d Out of memory\n",
                             __FUNCTION__, __LINE__);
             free(search_path);
             return VA_STATUS_ERROR_ALLOCATION_FAILED;
         }
 
-        va_infoMessage(dpy, "Trying to open %s\n", driver_path);
-#if defined(RTLD_NODELETE) && !defined(ANDROID)
-        handle = dlopen(driver_path, RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
-#else
-        handle = dlopen(driver_path, RTLD_NOW | RTLD_GLOBAL);
-#endif
-        if (!handle) {
-            /* Don't give errors for non-existing files */
-            if (0 == access(driver_path, F_OK))
-                va_errorMessage(dpy, "dlopen of %s failed: %s\n", driver_path, dlerror());
-        } else {
-            VADriverInit init_func = NULL;
-            char init_func_s[256];
-            int i;
-
-            struct {
-                int major;
-                int minor;
-            } compatible_versions[VA_MINOR_VERSION + 2];
-            for (i = 0; i <= VA_MINOR_VERSION; i ++) {
-                compatible_versions[i].major = VA_MAJOR_VERSION;
-                compatible_versions[i].minor = VA_MINOR_VERSION - i;
-            }
-            compatible_versions[i].major = -1;
-            compatible_versions[i].minor = -1;
-
-            for (i = 0; compatible_versions[i].major >= 0; i++) {
-                if (va_getDriverInitName(init_func_s, sizeof(init_func_s),
-                                         compatible_versions[i].major,
-                                         compatible_versions[i].minor)) {
-                    init_func = (VADriverInit)dlsym(handle, init_func_s);
-                    if (init_func) {
-                        va_infoMessage(dpy, "Found init function %s\n", init_func_s);
-                        break;
-                    }
-                }
-            }
-
-            if (compatible_versions[i].major < 0) {
-                va_errorMessage(dpy, "%s has no function %s\n",
-                                driver_path, init_func_s);
-                dlclose(handle);
-            } else {
-                struct VADriverVTable *vtable = ctx->vtable;
```

Comment:
Side note: moving this massive else statement (or portion of it) to it's own function has merit in itself.